### PR TITLE
refactor: modularize cli utilities

### PR DESCRIPTION
**Pull request by felangel**
## Description

- refactor: modularize cli utilities
  - preparation for adding a Dart CLI which supports `dart fix`

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [X] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by marcossevilla** [comment]
@felangel what if we use barrel files for a "CLI feature" instead of using `part of`, as they might be more CLIs in the future and might be easier to maintain as we would have imports specific to one CLI on each file?

Either approach looks good to me but just had that thought. 🤔

**Comment by felangel** [comment]
> @felangel what if we use barrel files for a "CLI feature" instead of using `part of`, as they might be more CLIs in the future and might be easier to maintain as we would have imports specific to one CLI on each file?
> 
> Either approach looks good to me but just had that thought. 🤔

The reason I went with this approach is to keep `_Cmd` private. If we went with a barrel we'd need to publicly expose the `Cmd` class.

File: lib/src/cli/cli.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import 'package:path/path.dart' as p;
+import 'package:universal_io/io.dart';
+
+part 'flutter_cli.dart';
+
+/// Abstraction for running commands via command-line.
+class _Cmd {
+  /// Runs the specified [cmd] with the provided [args].
+  static Future<ProcessResult> run(
```
Comment (orestesgaolin) [inline lib/src/cli/cli.dart:9]: Are you fully convinced to run commands via static methods? Maybe having a CommandClient or something similar would be more inline with our general approach to modularization?

Comment (felangel) [reply lib/src/cli/cli.dart:9]: Yeah that's a larger refactor we can prioritize. My primary goal for this PR was to make it so we could easily introduce a `dart_cli.dart` to support `dart fix`. The existing `_Cmd` API is not affected by these changes but I think we can create a separate PR to address that. wdyt?

Comment (orestesgaolin) [reply lib/src/cli/cli.dart:9]: sounds good, just wanted to chime in but I'm ok with the current approach

**Review by orestesgaolin — APPROVED** [review]

**Pull request merged**